elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -139,6 +167,20 @@ private RealmAddHandler() {
             super(SECURITY_REALM_RUNTIME_CAPABILITY, ATTRIBUTES);
         }
 
+        private SecretKey getKey(OperationContext context, String credentialStoreReference, String alias) throws OperationFailedException{
+            ExceptionFunction<OperationContext, CredentialStore, OperationFailedException> credentialStoreApi = context.getCapabilityRuntimeAPI(CREDENTIAL_STORE_API_CAPABILITY, credentialStoreReference, ExceptionFunction.class);
+            CredentialStore credentialStoreResource = credentialStoreApi.apply(context);
+            try {
+                SecretKeyCredential credential = credentialStoreResource.retrieve(alias, SecretKeyCredential.class);
+                if (credential == null) {
+                    throw ROOT_LOGGER.credentialDoesNotExist(alias, "SecretKey");
+                }
+                return credential.getSecretKey();
+            } catch (CredentialStoreException e) {
+                throw ROOT_LOGGER.unableToLoadCredential(e);
```

Comment:
Looks like `unableToLoadCredential` is a message that was created for expression resolution (notice the type is `ExpressionResolver.ExpressionResolutionUserException`). Would be good to use a different one here.

Reply:
Added new error: `unableToLoadCredentialStore`
https://github.com/wildfly/wildfly-core/blob/fd0416c917a0a599199f1c39d6a9dddf5a17080a/elytron/src/main/java/org/wildfly/extension/elytron/_private/ElytronSubsystemMessages.java#L693-L694